xchange-core/src/main/java/org/knowm/xchange/currency/Currency.java

```diff
@@ -472,11 +482,22 @@ public boolean equals(Object obj) {
   public int compareTo(Currency o) {
 
     if (attributes.equals(o.attributes)) return 0;
+    if (attributes.commonCode == null) {
+		return o.attributes.commonCode == null ? 0 : -1;
+    }
```

Comment:
Why the rough words? I am not working for you know...Next time just ease things and write your suggested solution instead of doing code shaming...

Anyways, fixed.

Reply:
Sorry for the rough language but your pushing a PR with many unrelated changes that are shallowly tested and would have broken many applications that are using this library. No one is working for anyone here you know - I'm not getting paid for this review. Its better to have working code then new buggy features and not taking the bigger picture into account does not help.